athena-neptune/src/main/java/com/amazonaws/athena/connectors/neptune/NeptuneMetadataHandler.java

```diff
@@ -243,4 +271,63 @@ protected Field convertField(String name, String glueType)
     {
         return GlueFieldLexer.lex(name, glueType);
     }
+
+    @Override
+    public GetTableResponse doGetQueryPassthroughSchema(BlockAllocator allocator, GetTableRequest request) throws Exception
+    {
+        Map<String, String> qptArguments = request.getQueryPassthroughArguments();
+        queryPassthrough.verify(qptArguments);
+        String schemaName = qptArguments.get(NeptuneQueryPassthrough.DATABASE);
+        String tableName = qptArguments.get(NeptuneQueryPassthrough.COLLECTION);
+        TableName tableNameObj = new TableName(schemaName, tableName);
+        request = new GetTableRequest(request.getIdentity(), request.getQueryId(),
+                request.getCatalogName(), tableNameObj, request.getQueryPassthroughArguments());
+
+        GetTableResponse getTableResponse = doGetTable(allocator, request);
+        List<Field> fields = getTableResponse.getSchema().getFields();
+        Client client = neptuneConnection.getNeptuneClientConnection();
+        GraphTraversalSource graphTraversalSource = neptuneConnection.getTraversalSource(client);
+        String gremlinQuery = qptArguments.get(NeptuneQueryPassthrough.QUERY);
+        gremlinQuery = gremlinQuery.concat(".limit(1)");
+        logger.info("NeptuneMetadataHandler doGetQueryPassthroughSchema gremlinQuery with limit: " + gremlinQuery);
+        ScriptEngine engine = new GremlinGroovyScriptEngine();
+        Bindings bindings = engine.createBindings();
+        bindings.put("g", graphTraversalSource);
+        Object object = engine.eval(gremlinQuery, bindings);
+        GraphTraversal graphTraversalForSchema = (GraphTraversal) object;
+        Map graphTraversalObj = null;
+        Schema schema;
+        while (graphTraversalForSchema.hasNext()) {
```

Comment:
Do we really need this? It seems like we just traversing here without any ops?

Reply:
Or we just need the left (end node)?

Reply:
Hi @chngpe Yes, it's required, here we are preparing map as 'graphTraversalObj' and using keys from map to construct schema builder at line no.306